cip/1.accepted/CIP2017-03-29-Single-Value-Subqueries.adoc

```diff
@@ -0,0 +1,68 @@
+= CIP2017-03-29 Scalar Subqueries and List Subqueries
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Tobias Lindaaker <tobias.lindaaker@neotechnology.com>
+
+toc::[]
+
+== Scalar Subqueries and List Subqueries
+
+Scalar Subqueries are read-only subqueries that produce a single value in a single row.
+The result of a Scalar Subquery is the single value (in the single row) produced by the subquery.
+
+List Subqueries are read-only subqueries that produce a single value per row, and zero or more rows.
+The result of a List Subquery is to collect the value of all rows produced by the subquery into a list.
+
+=== Syntax
+
+[source, ebnf]
+----
+ListSubquery = '[', SingleValueSubquery, ']' ;
+
+ScalarSubquery = '<(((<', SingleValueSubquery, '>)))>' ;
```

Comment:
More random ideas:

- `VALUE ( subquery )` (but: too generic?)
- `QUERY ( subquery )` (but: too generic?
- `EVAL ( subquery )` (but: sounds too much like metaprogramming?)
- `$( subquery )` (but: feels like we should reserve this syntax, perhaps for metaprogramming/templating)
- `(| ()-[]->(c) RETURN c |)` - Bananas ftw :)